servers/rendering/shader_language.h

```diff
@@ -720,6 +720,26 @@ class ShaderLanguage {
 				Node(TYPE_SHADER) {}
 	};
 
+	struct UniformOrderComparator {
+		_FORCE_INLINE_ bool operator()(const Pair<StringName, ShaderNode::Uniform> &A, const Pair<StringName, ShaderNode::Uniform> &B) const {
+			int left = 0;
+			if (A.second.texture_order >= 0) {
+				left = A.second.texture_order + 100000;
+			} else {
+				left = A.second.order;
+			}
```

Comment:
I would simplify this by making the ``second`` property an int like it used to be so that the comparison can be very simple. Then, like in the old system, I would set the value of ``second`` based on ``order`` or ``texture_order`` as appropriate. 

I.e. instead of:
```
filtered_uniforms.push_back(Pair<StringName, ShaderLanguage::ShaderNode::Uniform>(E.key, E.value));
```
Do
```
if (E.value.texture_order >= 0) {
	filtered_uniforms.push_back(Pair<StringName, int>(E.key, E.value.texture_order + 100000));
} else {
	filtered_uniforms.push_back(Pair<StringName, int>(E.key, E.value.order));
}
```

Then here you can just do:
```
return A.second < B.second;
```

Reply:
Thanks, Done.